mobile-app/src/components/habit-components/HabitCard.tsx

```diff
@@ -20,7 +26,7 @@ export function HabitCard({ habitId }: { habitId: string }) {
     <Link
       push
       href={{
-        pathname: "/viewhabit",
+        pathname: screenType.concat("/viewhabit"),
```

Comment:
And then here just do a check

pathname: openAsModal? "modals/viewhabit" : "viewhabit"